[RHACS] ROX-20866: short lived token integrations (#72100)
openshift/openshift-docs: modules/google-workload-identity-federation.adoc

// Module included in the following assemblies:
//
// * integration/integrate-using-short-lived-tokens.adoc
:_mod-docs-content-type: PROCEDURE
[id="google-workload-identity-federation_{context}"]

> = Google workload identity federation

Reviewer: Should be a gerund since it's a procedure https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-procedures

{product-title-short} integrations may authenticate against Google Cloud Platform using link:https://cloud.google.com/iam/docs/workload-identity-federation[workload identities]. To enable workload identity authentication in a Google Cloud integration, select the *Use workload identity* upon creation.

[IMPORTANT]
====

> The Google service account that is associated to the {product-title-short} pod via the workload identity must have the IAM permissions required by the integration. For example, to set up a workload identity for integrating with Google Artifact Registry, connect a service account with the *roles/artifactregistry.reader* role. For more information on Google IAM roles see link:https://cloud.google.com/artifact-registry/docs/access-control[Configure roles and permissions].

Reviewer: See guidance above on via.

====

> == Google Kubernetes Engine (GKE)

Reviewer: Needs ID/gerund.

.Prerequisites
* Access to the Google Cloud project containing the cluster and integration resources.

.Procedure
. Follow the Google Cloud Platform documentation on how to link:https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity[Use workload identity federation for GKE].

> . Annotate the {product-title-short} service account.

Reviewer: ```suggestion
. Annotate the {product-title-short} service account by running the following command: 
```

> +

Reviewer: Watching spacing here.

[source,terminal]
----
$ kubectl annotate serviceaccount central \ <1>
    --namespace stackrox \
    iam.gke.io/gcp-service-account=<GSA_NAME>@<GSA_PROJECT>.iam.gserviceaccount.com
----
<1> When setting up delegated scanning, choose *sensor* instead of *central*.

> == OpenShift

Reviewer: ID/gerund.

.Prerequisites

> * An OpenShift cluster with the link:https://docs.openshift.com/container-platform/4.14/authentication/managing_cloud_provider_credentials/cco-short-term-creds.html[Cloud Credential Operator in manual mode for short-term credentials]. In particular, you will need a public OIDC configuration bucket with the OpenShift service account signer key.

Reviewer: ```suggestion
* An {ocp} cluster with the link:https://docs.openshift.com/container-platform/4.14/authentication/managing_cloud_provider_credentials/cco-short-term-creds.html[Cloud Credential Operator in manual mode for short-term credentials]. You must have a public OIDC configuration bucket with the OpenShift service account signer key.
```

* Access to Google Cloud project with the *roles/iam.workloadIdentityPoolAdmin* role.

.Procedure

> . Follow link:https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#iam-workload-pools-list-gcloud[Manage workload identity pools] to create a workload identity pool. For example:

Reviewer: ```suggestion
. Use the link:https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#iam-workload-pools-list-gcloud[Manage workload identity pools] instructions to create a workload identity pool. For example:
```

+
[source,terminal]
----
$ gcloud iam workload-identity-pools create rhacs-pool \
    --location="global" \
    --display-name="RHACS workload pool"
----

> . Follow link:https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#manage-providers[Manage workload identity pool providers] to create a workload identity pool provider. For example:

Reviewer: ```suggestion
. Use the link:https://cloud.google.com/iam/docs/manage-workload-identity-pools-providers#manage-providers[Manage workload identity pool providers] instructions to create a workload identity pool provider. For example:
```

+
[source,terminal]
----
$ gcloud iam workload-identity-pools providers create-oidc rhacs-provider \
    --location="global" \
    --workload-identity-pool="rhacs-pool" \
    --display-name="RHACS provider" \
    --attribute-mapping="google.subject=assertion.sub" \

>     --issuer-uri="https://<OIDC-configuration-URL>" \

Reviewer: ```suggestion
    --issuer-uri="https://<OIDC_configuration_URL>" \
```

    --allowed-audiences=openshift
----

> . Connect a Google service account to the workload identity pool. For example:

Reviewer: ```suggestion
. Enter the following command to connect a Google service account to the workload identity pool. For example:
```
